### PR TITLE
FIX-#6110: Generalize logic to test if an index is a MultiIndex

### LINKED ISSUE #6110: REFACTOR: check number of levels in index with nlevels instead of looking for pd.MultiIndex
porting https://github.com/ponder-org/modin-public/commit/40932f3586270454f0512a20cc41ecda9b3b4f04

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6110 ? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/series.py [hunk 1]
```diff
@@ -1425,9 +1425,13 @@ def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
...
-        result = DataFrame(
-            query_compiler=self._query_compiler.unstack(level, fill_value)
-        )
+        # We can't unstack a Series object, if we don't have a MultiIndex.
+        if len(self.index.names) > 1:
+            result = DataFrame(
+                query_compiler=self._query_compiler.unstack(level, fill_value)
+            )
+        else:
+            raise ValueError(f"index must be a MultiIndex to unstack, {type{obj.index}} was passed")
```
Comment (RehanSD) [inline modin/pandas/series.py:1434]: Thinking about adding a `pandas_name` attribute to our ponder Indexes, and using that so we can get something kind of similar to what users would expect for this error. Can always revert it back to not showing type as is done in Modin-public.

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -1523,6 +1523,9 @@ def pivot(self, index=None, columns=None, values=None):  # noqa: PR01, RT01, D20
         """
         Return reshaped ``DataFrame`` organized by given index / column values.
         """
+        # If values is None, we use all remaining columns not in index or columns.
+        if values is None:
+            values = [c for c in self.columns if c not in index and c not in columns]
```
Comment (vnlitvinov) [inline modin/pandas/dataframe.py:1528]: I wonder how this matches with PR title... seems like an unrelated change.

**Review by vnlitvinov — COMMENTED** [review]
@RehanSD CI is failing

File: modin/pandas/dataframe.py (hunk 2 above)
Comment (mvashishtha) [reply modin/pandas/dataframe.py:1528]: yes, we should revert this change. Let's move it to https://github.com/modin-project/modin/issues/6146.

File: modin/pandas/dataframe.py [hunk 3]
```diff
@@ -1506,10 +1506,10 @@ def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
         """
         Pivot a level of the (necessarily hierarchical) index labels.
         """
-        if not isinstance(self.index, pandas.MultiIndex) or (
-            isinstance(self.index, pandas.MultiIndex)
-            and is_list_like(level)
-            and len(level) == self.index.nlevels
+        # This ensures that non-pandas MultiIndex objects are caught.
```
[the remark this answers is not in the document]
Comment (devin-petersohn) [reply modin/pandas/dataframe.py:1509]: It's more of a check that removes assumptions from the API layer. The API layer should not be assuming that every MultiIndex is a pandas MultiIndex.
